nuxeo-services/login/nuxeo-platform-login-saml2/src/test/java/org/nuxeo/ecm/platform/auth/saml/SAMLAuthenticatorWithMapperTest.java

```diff
@@ -59,38 +60,50 @@
 import java.io.InputStream;
 import java.net.URISyntaxException;
 import java.net.URL;
+import java.util.HashMap;
 import java.util.Map;
 import java.util.zip.Inflater;
 import java.util.zip.InflaterInputStream;
 
 @RunWith(FeaturesRunner.class)
 @Features({ UserMapperFeature.class, DirectoryFeature.class })
 @RepositoryConfig(init = DefaultRepositoryInit.class, cleanup = Granularity.METHOD)
-@Deploy({ "org.nuxeo.ecm.platform.login.saml2" })
-@LocalDeploy({"org.nuxeo.ecm.platform.auth.saml:OSGI-INF/test-sql-directory.xml","org.nuxeo.ecm.platform.auth.saml:OSGI-INF/usermapper-contribs.xml"})
+@Deploy({ "org.nuxeo.ecm.platform.usermanager", "org.nuxeo.ecm.platform.login.saml2" })
+@LocalDeploy({ "org.nuxeo.ecm.platform.auth.saml:OSGI-INF/test-sql-directory.xml" })
 public class SAMLAuthenticatorWithMapperTest {
 
     @Inject
     protected UserManager userManager;
 
     private SAMLAuthenticationProvider samlAuth;
 
+    private void initAuthProvider(Map<String, String> params) throws URISyntaxException {
+    	String metadata = getClass().getResource("/idp-meta.xml").toURI().getPath();
+    	Map<String, String> newParams = new ImmutableMap.Builder<String, String>()
+    	        .put("metadata", metadata)
+    	        .putAll(params == null ? new HashMap<String, String>() : params)
+    	        .build();
+    	
+    	samlAuth = new SAMLAuthenticationProvider();
+    	samlAuth.initPlugin(newParams);
+    }
+    
+    private Map<String, String> getParamMap(boolean createIfNeeded, boolean update) {
+    	Map<String, String> resultMap = new HashMap<String, String>();
+    	resultMap.put("userResolverCreateIfNeeded", new Boolean(createIfNeeded).toString());
+    	resultMap.put("userResolverUpdate", new Boolean(update).toString());
+    	
+    	return resultMap;
+    }
+    
     @Before
     public void doBefore() throws URISyntaxException {
```

Comment:
No use for the `doBefore` if you're always reinitializing the params differently in each test case